src/components/dashboard/ReceiveConfirmation.js

```diff
@@ -23,10 +27,11 @@ export type ReceiveProps = {
 
 const ReceiveConfirmation = ({ screenProps, styles, ...props }: ReceiveProps) => {
   const profile = GDStore.useStore().get('profile')
+  const [showErrorDialog] = useErrorDialog()
   const [screenState] = useScreenState(screenProps)
   const { amount, code, reason, counterPartyDisplayName } = screenState
   const share = useMemo(() => {
-    if (canShare) {
+    if (canShare()) {
```

Comment:
i see that sharing logic is used al least in two components.

we could write an custom hook useNativeSharing() which will return canShare status and helper functions such as generateCode, generateReceiveShareObject, generateShareLink

Reply:
done